File: 251119/블럭쌓는 명령2/block-stacking-commands2.py
n, k = map(int, input().split())
commands = [tuple(map(int, input().split())) for _ in range(k)]

blocks = [0 for _ in range(n + 1)]
for s, e in commands:
    for i in range(s, e + 1):
        blocks[i] += 1

print(max(blocks))

File: 260308/블럭쌓는 명령2/block-stacking-commands2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const [n, k] = input[0].split(' ').map(Number);
const segments = input.slice(1, k + 1).map(line => line.split(' ').map(Number));

let seq = Array(n).fill(0);

for ([start, end] of segments) {
    for (let i = start; i <= end; i++) {
        seq[i] += 1;
    }
}

console.log(Math.max(...seq))